src/v/cluster/rm_stm.cc

```diff
@@ -1412,6 +1412,15 @@ static void filter_intersecting(
 
 ss::future<std::vector<rm_stm::tx_range>>
 rm_stm::aborted_transactions(model::offset from, model::offset to) {
+    return _state_lock.hold_read_lock().then(
```

Comment:
With this patch we are aggressively cleaning up aborted idx state,  do we need to make sure that the archiver has successfully uploaded them to cloud storage before we clean them up (#5381)? That synchronization was not obvious to me, so just want to make sure.

Reply:
as I understand the segment isn't removed before it's uploaded and they upload a segment along with aborted transactions, so things should be safe, @Lazin can you confirm it?

Reply:
Confirm. If the segment is deleted it's safe to remove aborted transactions state for its offset range.